src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -654,7 +654,7 @@ private bool InferConnectionDetails()
                 Port = port;
             }
             // Instance Name Handling. 
```

Comment:
I'd revert the comments, too.
```suggestion
            // Instance Name Handling. Only if we found a '\' and we did not find a port in the Data Source
```